### ISSUE TITLE
Nested User fields do not exist

### ISSUE THREAD

**Issue by AaronBuxbaum** [abraham/twitter-d#40]
Under `Status.User`, any key that is not in _both_ `User` and `TrimmedUser` does not exist. I believe this is because of the incorrect type here:
```
export type User = TrimmedUser | FullUser
```

<img width="555" alt="Screen Shot 2019-07-26 at 2 55 49 AM" src="https://user-images.githubusercontent.com/5578581/61932261-e3774600-af50-11e9-9daa-4090b278d36e.png">

**Comment by abraham** [comment]
`truncated` is not a property that would exist on either of those. You probably mean to be getting that off of a `Status` type.

**Comment by AaronBuxbaum** [comment]
Hurp... that was a bad example. `screen_name` is a better example of a key that's in FullUser but not TrimmedUser.

**Comment by abraham** [comment]
I've added a new helper method `isFullUser` and published v0.3.1 so you can now use a typeguard that you have the correct User type.

```typescript
import { isFullUser, User } from 'twitter-d';

function logUser(user: User) {
  if (isFullUser(user)) {
    console.log(user.name);
  } else {
    throw new Error('User is not type FullUser');
  }
}
```

**Comment by AaronBuxbaum** [comment]
nice, this is perfect. Appreciate the swift work!
